src/test/record/records_service/test_range.py

```diff
@@ -0,0 +1,100 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from caret_analyze.record import ColumnValue, Range
+from caret_analyze.record.record_factory import RecordsFactory
+
+
+def create_records(records_raw):
+    columns = [ColumnValue('start'), ColumnValue('end')]
+    records = RecordsFactory.create_instance()
+    for column in columns:
+        records.append_column(column, [])
+
+    for record_raw in records_raw:
+        records.append(record_raw)
+    return records
+
+
+class TestRange:
+
+    def test_get_range_normal(self, mocker):
+        records_raw1 = [
+            {'start': 0, 'end': 2},
+            {'start': 3, 'end': 4},
+            {'start': 11, 'end': 12}
+        ]
+        records_raw2 = [
+            {'start': 1, 'end': 3},
+            {'start': 4, 'end': 5},
+            {'start': 12, 'end': 13}
+        ]
+        records_list = [create_records(records_raw1), create_records(records_raw2)]
+        record_range = Range(records_list)
+        assert record_range.get_range() == (0, 12)
+
+    def test_get_range_drop_other_column(self, mocker):
+        records_raw1 = [
+            {'start': 0, 'end': 2},
+            {'start': 3, 'end': 4},
+            {'start': 11}
+        ]
+        records_raw2 = [
+            {'start': 1, 'end': 3},
+            {'start': 4, 'end': 5},
+            {'start': 12}
+        ]
+        records_list = [create_records(records_raw1), create_records(records_raw2)]
+        record_range = Range(records_list)
+        assert record_range.get_range() == (0, 12)
+
+    def test_get_range_drop_target_column(self, mocker):
+        records_raw1 = [
+            {'end': 2},
+            {'start': 3, 'end': 4},
+            {'start': 11, 'end': 12}
+        ]
+        records_raw2 = [
+            {'start': 1, 'end': 3},
+            {'start': 4, 'end': 5},
+            {'end': 13}
+        ]
+        records_list = [create_records(records_raw1), create_records(records_raw2)]
+        record_range = Range(records_list)
+        assert record_range.get_range() == (1, 11)
+
+    def test_get_range_exist_empty_records(self, mocker):
+        records_raw1 = [
+            {'start': 0, 'end': 2},
+            {'start': 3, 'end': 4},
+            {'start': 11, 'end': 12}
+        ]
+        records_raw2 = [
+        ]
+        records_list = [create_records(records_raw1), create_records(records_raw2)]
+        record_range = Range(records_list)
+        assert record_range.get_range() == (0, 11)
+
+    def test_get_range_no_input(self, mocker):
+        record_range = Range([])
+        assert record_range.get_range() == (0, 1)
+
+    def test_get_range_only_empty_records(self, mocker):
+        records_raw1 = [
+        ]
+        records_raw2 = [
+        ]
+        records_list = [create_records(records_raw1), create_records(records_raw2)]
+        record_range = Range(records_list)
+        assert record_range.get_range() == (0, 1)
```

Comment:
You need to test if `logger.warning('No valid measurement data.')` is occurred.
`test_get_range_no_input`, too.

Reply:
Addressed in https://github.com/tier4/CARET_analyze/pull/254/commits/827f7df21d4ddda2f02ece9f382020ba07940142.